### PR TITLE
feat: allow encoding of borrowed data

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic
 Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.7.0]
+
+### Changed
+
+- Use GATs for `Encoder` trait to allow encoding of borrowed data.
+  See [PR x](https://github.com/mxinden/asynchronous-codec/pull/x).
```
Comment (thomaseizinger) [inline CHANGELOG.md:14]:
```suggestion
  See [PR 9](https://github.com/mxinden/asynchronous-codec/pull/9).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic
 Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.7.0]
+
+### Changed
+
+- Use GATs for `Encoder` trait to allow encoding of borrowed data.
+  See [PR 9](https://github.com/mxinden/asynchronous-codec/pull/9).
```
Comment (mxinden) [inline CHANGELOG.md:14]: What do you think of including the diff to the `LengthCodec` below here to help folks upgrade to the new version?

**Review by mxinden — COMMENTED** [review]
Fancy!

File: CHANGELOG.md (hunk 2 above)
Comment (thomaseizinger) [reply CHANGELOG.md:14]: Are the docs and the link to the PR not enough?

**Comment by thomaseizinger** [comment]
Friendly ping @mxinden.

File: CHANGELOG.md (hunk 2 above)
Comment (mxinden) [reply CHANGELOG.md:14]: It reduces one step for users.I think it might safe them some time. Not a strong preference.

Comment (thomaseizinger) [reply CHANGELOG.md:14]: Would prefer to just merge this to be honest :)

**Review by mxinden — APPROVED** [review]

**Comment by mxinden** [comment]
[Tagged](https://github.com/mxinden/asynchronous-codec/releases/tag/v0.7.0) and [published](https://crates.io/crates/asynchronous-codec/0.7.0). Thank you @thomaseizinger.

**Pull request merged**